### PR TITLE
Make it possible to not specify clang binary in bindgen

### PR DESCRIPTION
**Pull request by hlopko**

### CODE REVIEW

File: bindgen/private/bindgen.bzl [hunk 1]
```diff
@@ -201,13 +201,13 @@ def _rust_bindgen_impl(ctx):
...
 
-    tools = depset([clang_bin], transitive = [cc_toolchain.all_files])
+    tools = depset(([clang_bin] if clang_bin else []), transitive = [cc_toolchain.all_files])
 
     # libclang should only have 1 output file
     libclang_dir = _get_libs_for_static_executable(libclang).to_list()[0].dirname
 
     env = {
-        "CLANG_PATH": clang_bin.path,
+        "CLANG_PATH": (clang_bin.path if clang_bin else "/dev/null"),
```
Comment (krasimirgg) [inline bindgen/private/bindgen.bzl:210]: Does this still work if we just omit the CLANG_PATH key in this case? (If yes, plz do that as it a bit cleaner)

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: bindgen/private/bindgen.bzl (hunk 1 above)
Comment (hlopko) [reply bindgen/private/bindgen.bzl]: Done.

File: bindgen/private/bindgen.bzl [hunk 2]
```diff
@@ -201,16 +201,17 @@ def _rust_bindgen_impl(ctx):
...
     # libclang should only have 1 output file
     libclang_dir = _get_libs_for_static_executable(libclang).to_list()[0].dirname
 
     env = {
-        "CLANG_PATH": clang_bin.path,
         "LIBCLANG_PATH": libclang_dir,
         "RUST_BACKTRACE": "1",
     }
+    if clang_bin:
+        env["clang_path"] = clang_bin.path
```
Comment (krasimirgg) [inline bindgen/private/bindgen.bzl:214]: shouldn't this be capitalized?

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: bindgen/private/bindgen.bzl (hunk 2 above)
Comment (hlopko) [reply bindgen/private/bindgen.bzl]: /o\ yep..

**Review by krasimirgg — APPROVED** [review]
Thank you!

**Review by krasimirgg — APPROVED** [review]
Thank you!

**Pull request merged**